pkg/services/authz/legacy/client/client.go

```diff
@@ -0,0 +1,227 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/grpc"
+)
+
+type CheckRequest struct {
+	// Namespace is either `org-<id>` or `stacks-<id>`.
+	Namespace string
+
+	// Subject is the typed identity we want to perform a check for
+	Subject string
+
+	// Action to check access for
+	Action string
+
+	// Resource is ~Kind eg dashboards
+	Resource string
+	// Attribute used to store permission
+	Attr string
+	// Name is the identifier for the resource.
+	// In grafana, this was historically called "UID", but in k8s, it is the name
+	Name string
+
+	// Contextuals are additional resource + name that should be checked.
+	// E.g. for dashboards this can be the folder that a it belong to.
+	Contextuals []Contextual
+}
+
+type CheckResponse struct {
+	Allowed bool
+}
+
+type Contextual struct {
+	// Resource is ~Kind eg dashboards
+	Resource string
+	// Attribute used to store permission
+	Attr string
+	// Name is the identifier for the resource.
+	// In grafana, this was historically called "UID", but in k8s, it is the name
+	Name string
+}
+
+type ListRequest struct {
+	// Namespace is either `org-<id>` or `stacks-<id>`.
+	Namespace string
+	// Subject is the typed identity we want to perform a check for
+	Subject string
+	// Action to check access for
+	Action string
+	// Resource is ~Kind eg dashboards
+	Resource string
+	// Attribute used to store permission
+	Attr string
+}
+
+// Checks access while iterating within a resource
+type ItemChecker func(name string, extra ...Contextual) bool
+
+// FIME: this is the "new" client interface?
+type ReadClient interface {
```

Comment:
Do we need to implement `List()` as well?

Reply:
This is not the final client but just something to test with. List is something we would want. But if we decide to go with this I will probably slim down this client a bit and we can work on the details in a separate pr.

Reply:
Btw for now I will remove everything except `Check` so that there is not to much to review in this one